integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/GCPUtils.java

```diff
@@ -0,0 +1,251 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.agent.util;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.google.common.io.CharStreams;
+import io.openlineage.client.Environment;
+import io.openlineage.spark.api.OpenLineageContext;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.http.Consts;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.apache.spark.SparkContext;
+import org.apache.spark.sql.execution.SparkPlan;
+import org.apache.spark.sql.execution.WholeStageCodegenExec;
+
+/** Util to extract values from GCP environment */
+public class GCPUtils {
```

Comment:
I would appreciate writing some tests for this class. You can start MockServer that responds with the jsons and verify if the class methods work as expected.

Reply:
Okay, working on this.

Reply:
Hi @pawel-big-lebowski, I have added tests for this in the recent commit. Please have a look whenever convenient.